examples/java/core/polymorphic_container/src/main/java/com/microsoft/bond/examples/PolymorphicContainer.java

```diff
@@ -0,0 +1,70 @@
+package com.microsoft.bond.examples;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import com.microsoft.bond.Bonded;
+import com.microsoft.bond.Deserializer;
+import com.microsoft.bond.Serializer;
+
+import com.microsoft.bond.protocol.CompactBinaryReader;
+import com.microsoft.bond.protocol.CompactBinaryWriter;
+
+import com.microsoft.bond.examples.polymorphiccontainer.Circle;
+import com.microsoft.bond.examples.polymorphiccontainer.Polymorphic;
+import com.microsoft.bond.examples.polymorphiccontainer.Rectangle;
+import com.microsoft.bond.examples.polymorphiccontainer.Shape;
+import com.microsoft.bond.examples.polymorphiccontainer.Type;
+
+public class PolymorphicContainer {
+
+    public static void main(final String[] args) throws IOException {
+
+        final Circle circle = new Circle();
+        circle.Type = Type.Circle;
+        circle.Radius = 3.14;
+
+        final Rectangle rectangle = new Rectangle();
+        rectangle.Type = Type.Rectangle;
+        rectangle.Width = 10.0;
+        rectangle.Height = 5.5;
+
+        final Polymorphic src = new Polymorphic();
+        src.Shapes.add((Bonded<Shape>)Bonded.fromObject(circle, Circle.BOND_TYPE));
```

Comment:
These two lines are currently failing to compile. The first one, with the cast to `Bonded<Shape>` gives this error:
```
PolymorphicContainer.java:34: error: incompatible types: Bonded<Circle> cannot be converted to Bonded<Shape>
        src.Shapes.add((Bonded<Shape>)Bonded.fromObject(circle, Circle.BOND_TYPE));
```

The one without the cast gives this error:

```
PolymorphicContainer.java:35: error: no suitable method found for add(Bonded<Rectangle>)
        src.Shapes.add(Bonded.fromObject(rectangle, Rectangle.BOND_TYPE));
                  ^
    method Collection.add(Bonded<Shape>) is not applicable
      (argument mismatch; inferred type does not conform to equality constraint(s)
          inferred: Shape
          equality constraints(s): Shape,Rectangle)
    method List.add(Bonded<Shape>) is not applicable
      (argument mismatch; inferred type does not conform to equality constraint(s)
          inferred: Shape
          equality constraints(s): Shape,Rectangle)
```

My Java-fu is weak here: what should I be doing here instead?